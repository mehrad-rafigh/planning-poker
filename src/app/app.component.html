<a href="https://github.com/mehrad-rafigh/planning-poker"
  ><img
    width="149"
    height="149"
    src="https://github.blog/wp-content/uploads/2008/12/forkme_right_orange_ff7600.png?resize=149%2C149"
    class="attachment-full size-full"
    alt="Fork me on GitHub"
    data-recalc-dims="1"
/></a>
<router-outlet></router-outlet>
