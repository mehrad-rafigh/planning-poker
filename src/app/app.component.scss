a {
  position: absolute;
  z-index: 1;
  right: 0;
}
