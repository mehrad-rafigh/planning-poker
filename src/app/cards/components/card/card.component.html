<div
  class="card"
  [ngStyle]="{ 'background-color': cardColor.backgroundColor }"
  [ngClass]="setFontColor()"
  (click)="handleClick()"
>
  <span *ngIf="!detailMode" class="top-right quarter-h">{{ value }}</span>
  <span *ngIf="detailMode" class="half-h" [ngClass]="setDetailMode()">{{
    value
  }}</span>
  <span *ngIf="!detailMode" class="bottom-left quarter-h">{{ value }}</span>
</div>
