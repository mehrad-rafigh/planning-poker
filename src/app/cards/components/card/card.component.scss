@import "../../../../mixins";

span {
  font-weight: bold;
}

.card {
  display: flex;
  flex-flow: column nowrap;
  box-sizing: border-box;
  height: 200px;
  width: auto;
  border-color: lightslategray;
  position: relative;
  box-shadow: 0 4px 8px 0 rgba(0, 0, 0, 0.2);
  transition: 0.3s;
  border-radius: 15px;

  @include mobile() {
    height: 250px;
    font-size: 26px;
  }

  @include tablet() {
    height: 300px;
    font-size: 24px;
  }

  @include above-tablet() {
    height: 350px;
    font-size: 22px;
  }
}

.top-right {
  text-align: right;
  margin-top: 32px;
  margin-right: 32px;
}

.bottom-left {
  margin-bottom: 32px;
  margin-left: 32px;
  position: absolute;
  bottom: 0;
  height: auto !important;
}

.quarter-h {
  height: 25%;
}

.half-h {
  height: 50%;
}

.detail-mode {
  height: unset;
  margin: auto;
  font-size: 128px;
}

.font-white {
  color: white;
}
