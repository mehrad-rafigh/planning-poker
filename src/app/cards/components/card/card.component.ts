import {
  Component,
  OnInit,
  ChangeDetectionStrategy,
  Input,
  EventEmitter,
  Output
} from "@angular/core";

@Component({
  selector: "app-card",
  templateUrl: "./card.component.html",
  styleUrls: ["./card.component.scss"],
  changeDetection: ChangeDetectionStrategy.OnPush
})
export class CardComponent implements OnInit {
  @Input() value: string;
  @Input() cardColor: { backgroundColor: string; fontColor: string };
  @Input() detailMode = false;
  @Output() clickEvent = new EventEmitter<any>();
  constructor() {}

  ngOnInit(): void {}

  handleClick(): void {
    this.clickEvent.emit({
      value: this.value,
      card: {
        backgroundColor: this.cardColor.backgroundColor,
        fontColor: this.cardColor.fontColor
      }
    });
  }

  setDetailMode(): string {
    if (this.detailMode) {
      return "detail-mode";
    }
  }

  setFontColor(): string {
    if (this.cardColor.fontColor === "white") {
      return "font-white";
    }
  }
}
