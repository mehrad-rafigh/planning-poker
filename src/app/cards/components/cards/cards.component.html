<app-toolbar></app-toolbar>
<div class="cards-container">
  <app-card
    *ngFor="let no of range"
    [value]="no"
    [cardColor]="setCardColors()"
    (clickEvent)="showCardInDetail($event)"
  ></app-card>
</div>
