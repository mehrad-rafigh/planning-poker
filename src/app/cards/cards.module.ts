import { NgModule } from "@angular/core";
import { CommonModule } from "@angular/common";
import { CardsComponent } from "./components/cards/cards.component";
import { ToolbarModule } from "../toolbar/toolbar.module";
import { CardComponent } from "./components/card/card.component";
import { DetailComponent } from "./components/detail/detail.component";

@NgModule({
  declarations: [CardsComponent, CardComponent, DetailComponent],
  imports: [CommonModule, ToolbarModule],
  exports: [CardsComponent, DetailComponent]
})
export class CardsModule {}
