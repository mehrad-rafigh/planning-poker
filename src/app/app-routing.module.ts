import { NgModule } from "@angular/core";
import { Routes, RouterModule } from "@angular/router";
import { CardsComponent } from "./cards/components/cards/cards.component";
import { DetailComponent } from "./cards/components/detail/detail.component";

const routes: Routes = [
  {
    path: "",
    component: CardsComponent
  },
  {
    path: "detail",
    component: DetailComponent
  },
  {
    path: "settings",
    loadChildren: () =>
      import("../../projects/settings/src/lib/settings.module").then(
        module => module.SettingsModule
      )
  },
  {
    path: "**",
    component: CardsComponent
  }
];

@NgModule({
  imports: [
    RouterModule.forRoot(routes, {
      scrollPositionRestoration: "top"
    })
  ],
  exports: [RouterModule]
})
export class AppRoutingModule {}
