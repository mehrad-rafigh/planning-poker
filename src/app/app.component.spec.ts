import { async, ComponentFixture, TestBed } from "@angular/core/testing";
import { RouterTestingModule } from "@angular/router/testing";
import { AppComponent } from "./app.component";
import { routes } from "./app-routing.module";
import { Router } from "@angular/router";
import { CardsComponent } from "./cards/components/cards/cards.component";
import { DetailComponent } from "./cards/components/detail/detail.component";
import { CardComponent } from "./cards/components/card/card.component";
import { ToolbarComponent } from "./toolbar/components/toolbar/toolbar.component";
import { DebugElement } from "@angular/core";
import { By } from "@angular/platform-browser";

describe("AppComponent", () => {
  let router: Router;
  let component: AppComponent;
  let fixture: ComponentFixture<AppComponent>;
  let debugElement: DebugElement;

  beforeEach(async(() => {
    TestBed.configureTestingModule({
      imports: [RouterTestingModule.withRoutes(routes)],
      declarations: [
        AppComponent,
        CardsComponent,
        DetailComponent,
        CardComponent,
        ToolbarComponent
      ]
    }).compileComponents();
  }));

  beforeEach(() => {
    fixture = TestBed.createComponent(AppComponent);
    component = fixture.componentInstance;
    debugElement = fixture.debugElement;
    router = TestBed.inject<Router>(Router);

    router.initialNavigation();
    fixture.detectChanges();
  });

  it("should create the app component", () => {
    expect(component).toBeTruthy();
  });

  it("should match the snaphshot", () => {
    expect(fixture).toMatchSnapshot();
  });

  it("should render planning poker cards container", () => {
    const cards = debugElement.query(By.directive(CardsComponent));
    expect(cards).toBeTruthy();
  });
});
