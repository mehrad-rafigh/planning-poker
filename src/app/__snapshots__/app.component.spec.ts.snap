// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppComponent should match the snaphshot 1`] = `
<app-root>
  <a
    href="https://github.com/mehrad-rafigh/planning-poker"
  >
    <img
      alt="Fork me on GitHub"
      class="attachment-full size-full"
      data-recalc-dims="1"
      height="149"
      src="https://github.blog/wp-content/uploads/2008/12/forkme_right_orange_ff7600.png?resize=149%2C149"
      width="149"
    />
  </a><router-outlet /><app-cards>
    <app-toolbar>
      <nav
        class="navbar d-flex row-no-wrap"
      >
        <span
          class="navbar-title f-start"
        >
          Planning Poker
        </span>
      </nav>
    </app-toolbar>
    <div
      class="cards-container"
    >
      
    </div>
  </app-cards>
</app-root>
`;
